app/controllers/brapi/v1/germplasm_controller.rb

```diff
@@ -2,68 +2,64 @@ class Brapi::V1::GermplasmController < Brapi::BaseController
-      
+      germplasm_queries = Brapi::V1::GermplasmQueries.instance
+      page = get_page
+      page_size = get_page_size 
+          
+      result_object = germplasm_queries.germplasm_search_query(
+        params['germplasmPUI'],params['germplasmDbId'], params['germplasmName'], page, page_size, false)
       
-      if records.nil?
-        render json: { reason: 'Resource not found' }, status: :not_found
+      records = result_object.values
+     
+      if records.nil? || records.size ==0
+        render json: { reason: 'Resource not found' }, status: :not_found  # 404
       else
-        if records.size == 1 && records[0]!=nil && records[0] != api_key.user_id && records[1] 
-          render json: { reason: 'This is a private resource of another user' }, status: :unauthorized
-        else
-          json_result_array = []
-          
-          # any programmatic data manipulation can be done here
-          result_object.each do |row|
-            row = JSON.parse(row["row_to_json"])
-            puts row
-            if (!row[:user_id] || (row[:user_id]!=nil && row[:user_id] != api_key.user_id && row[:published] ))
-              json_result_array << row
-            end
-          end
-          
-          # pagination data returned
-          page = get_page
-          pageSize = get_pageSize 
+        
+        json_result_array = []
+        
+        # any programmatic data manipulation can be done here
+        result_object.each do |row|
+          row = JSON.parse(row["row_to_json"])
           
-          result_count_object = germplasm_query(params, true)
-          totalCount = result_count_object.values.first[0].to_i
-          totalPages = (totalCount/pageSize.to_f).ceil
-          
-          json_response = { 
-            'metadata' => json_metadata(pageSize, page, totalCount, totalPages),
-            'result' => json_result_array
-          }
-         
-          render json: json_response, :except => ["id", "user_id", "created_at", "updated_at","total_entries_count"]
+          # To check authentication and ownership when ORCID is supported by BrAPI
+          # We currently only retrieve public records. This is already done at query level
+          #if (!row[:user_id] || row[:published] )
+          json_result_array << row
+          #end
         end
-      end
+        
+        # pagination data returned
+        
+        result_count_object = germplasm_queries.germplasm_search_query(
+          params['germplasmPUI'],params['germplasmDbId'], params['germplasmName'], page, page_size, true)
+        total_count = result_count_object.values.first[0].to_i
+        total_pages = (total_count/page_size.to_f).ceil
+        
+        json_response = { 
+          'metadata' => json_metadata(page_size, page, total_count, total_pages),
+          'result' => json_result_array
+        }
+       
+        render json: json_response, :except => ["id", "user_id", "created_at", "updated_at", "total_entries_count"]
```

Comment:
Currently preferred hash syntax is `{ foo: "bar" }` instead of `{ :foo => "bar" }`.